### PR TITLE
Sdk 2264 net retrieve receipt

### PR DESCRIPTION
**Pull request by mehmet-yoti**

### CODE REVIEW

File: src/Yoti.Auth/DigitalIdentity/SharedReceiptResponse.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+namespace Yoti.Auth.DigitalIdentity
+{
+    public class SharedReceiptResponse
+    {
+        public string ID { get; set; }
+        public string SessionID { get; set; }
+        public string RememberMeID { get; set; }
+        public string ParentRememberMeID { get; set; }
+        public string Timestamp { get; set; }
+        public string Error { get; set; }
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/SharedReceiptResponse.cs:12]: is the `errorReason` not included because it is going to be added in another PR?

File: src/Examples/DigitalIdentity/DigitalIdentity/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,25 @@
+**/.classpath
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Examples/DigitalIdentity/DigitalIdentity/.dockerignore:1]: yes for docker files we need to ignore some defaults

File: src/Examples/DigitalIdentity/DigitalIdentity/Controllers/HomeController.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+                var yotiClient = new DigitalIdentityClient(_clientSdkId, privateKeyStream);
+
+
+                var ReceiptResult = yotiClient.GetShareReceipt(ReceiptID);
+
+                ViewBag.YotiClientSdkId = _clientSdkId;
+ 
+
+                return View("DigitalIdentity", ReceiptResult);
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Examples/DigitalIdentity/DigitalIdentity/Controllers/HomeController.cs:103]: solved

File: src/Examples/DigitalIdentity/DigitalIdentity/.dockerignore [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+**/charts
+**/docker-compose*
+**/Dockerfile*
+**/node_modules
+**/npm-debug.log
+**/obj
+**/secrets.dev.yaml
+**/values.dev.yaml
+LICENSE
+README.md
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Examples/DigitalIdentity/DigitalIdentity/.dockerignore:25]: ok, solved

File: src/Examples/DigitalIdentity/DigitalIdentity/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+version: '3.4'
+
+services:
+  coreexample:
+    image: ${DOCKER_REGISTRY-}coreexample
+    build:
+      context: ../../../
+      dockerfile: Examples/Profile/CoreExample/Dockerfile
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Examples/DigitalIdentity/DigitalIdentity/docker-compose.yml:8]: yes, corrected

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 6]
```diff
@@ -1,16 +1,27 @@
...
+using Yoti.Auth.ProtoBuf.Attribute;
+using Yoti.Auth.Share;
 
 namespace Yoti.Auth.DigitalIdentity
 {
     public static class DigitalIdentityService
     {
+        private const string identitySessionReceiptRetrieval = "/v2/receipts/{0}";
+        private const string identitySessionReceiptKeyRetrieval = "/v2/wrapped-item-keys/{0}";
+        private const string digitalIdentitySessionCreationEndpoint = "/v2/sessions";
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:22]: yes sure but in case in future i have set variable for future feature that can be added so we can use common variables. But i have simplified var names.

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 7]
```diff
@@ -31,8 +42,8 @@ internal static async Task<ShareSessionResult> CreateShareSession(HttpClient htt
                 .WithKeyPair(keyPair)
                 .WithBaseUri(apiUrl)
                 .WithHeader("X-Yoti-Auth-Id", sdkId)
```
[the remark this answers is not in the document]
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:44]: solved

File: src/Examples/DigitalIdentity/DigitalIdentity/.dockerignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (irotech) [reply src/Examples/DigitalIdentity/DigitalIdentity/.dockerignore]: u sure? If u look at the `Files chenged` tab most of the classes show a missing new line EOF

File: src/Examples/DigitalIdentity/DigitalIdentity/Controllers/HomeController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (irotech) [reply src/Examples/DigitalIdentity/DigitalIdentity/Controllers/HomeController.cs]: you sure u pushed the changes? I can still see multiple double empty lines

File: src/Examples/DigitalIdentity/DigitalIdentity/docker-compose.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (irotech) [reply src/Examples/DigitalIdentity/DigitalIdentity/docker-compose.yml]: I still see the old value

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (irotech) [reply src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs]: > But i have simplified var names.
Did u push?

File: src/Examples/DigitalIdentity/DigitalIdentity/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+WORKDIR /app
+EXPOSE 80
+EXPOSE 443
+
+FROM mcr.microsoft.com/dotnet/sdk:3.1 AS build
+WORKDIR /src
+COPY Examples/Profile/CoreExample/CoreExample.csproj Examples/Profile/CoreExample/
+COPY Yoti.Auth/Yoti.Auth.csproj Yoti.Auth/
+COPY . .
+WORKDIR /src/Examples/Profile/CoreExample
```
Comment (irotech) [inline src/Examples/DigitalIdentity/DigitalIdentity/Dockerfile:12]: this seems wrong as well

File: src/Yoti.Auth/DigitalIdentity/SharedReceiptResponse.cs (hunk 1 above)
Comment (irotech) [reply src/Yoti.Auth/DigitalIdentity/SharedReceiptResponse.cs:12]: have u got another ticket for that yet?

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 9]
```diff
@@ -159,5 +170,217 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+            {
+                if (!response.IsSuccessStatusCode)
+                {
+                    Response.CreateYotiExceptionFromStatusCode<DigitalIdentityException>(response);
+                }
+
+                var responseObject = await response.Content.ReadAsStringAsync();
+                var deserialized = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<ReceiptResponse>(responseObject));
+
+                return deserialized;
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:203]: unnecessary new line

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 10]
```diff
@@ -159,5 +170,217 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                        receiptResponse.OtherPartyContent.ExtraData,
+                        key);
+
+                ExtraData appExtraData = new ExtraData();
+                appExtraData = CryptoEngine.DecryptExtraData(
+                        receiptResponse.WrappedKey,
+                        receiptResponse.Content.ExtraData,
+                        key);
+
+
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:280]: double empty lines

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 11]
```diff
@@ -159,5 +170,217 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                if (!response.IsSuccessStatusCode)
+                {
+                    Response.CreateYotiExceptionFromStatusCode<DigitalIdentityException>(response);
+                }
+
+                var responseObject = await response.Content.ReadAsStringAsync();
+                var deserialized = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<ReceiptItemKeyResponse>(responseObject));
+
+                return deserialized;
+
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:340]: unnecessary empty line

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 12]
```diff
@@ -159,5 +170,217 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                        error = new Exception($"failed to decrypt receipt content extra data: {ex.Message}", ex);
+                        return (null, null, error);
+                    }
+                }
+            }
+
+            return (attrData, aextra, null);
+        }
+
+
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:384]: double empty lines

File: src/Yoti.Auth/CryptoEngine.cs [hunk 13]
```diff
@@ -132,5 +138,62 @@ internal static string GetAuthKey(AsymmetricCipherKeyPair keyPair)
...
+
+                return DecipherAes(receiptContentKey, decodedData.Iv.ToByteArray(), decodedData.CipherText.ToByteArray());
+            }
+            catch(Exception ex)
+            {
+                throw new Exception($"Failed to decrypt receipt content: {ex.Message}", ex);
+            }
+        }
     }
 }
```
Comment (irotech) [inline src/Yoti.Auth/CryptoEngine.cs:199]: EOF

File: src/Examples/DigitalIdentity/DigitalIdentity/wwwroot/static/profile.css [hunk 14]
```diff
@@ -0,0 +1,420 @@
...
+
+.yoti-edit-section {
+    padding: 50px 20px;
+}
+
+@media (min-width: 1100px) {
+    .yoti-edit-section {
+        padding: 75px 110px;
+    }
+}
```
Comment (irotech) [inline src/Examples/DigitalIdentity/DigitalIdentity/wwwroot/static/profile.css:420]: EOF

File: src/Examples/DigitalIdentity/DigitalIdentity/wwwroot/static/index.css [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+@media (min-width: 600px) {
+    .yoti-store-buttons-section {
+        grid-template-columns: 1fr 1fr;
+        grid-gap: 25px;
+    }
+}
+
+.yoti-app-button-link {
+    text-decoration: none;
+}
```
Comment (irotech) [inline src/Examples/DigitalIdentity/DigitalIdentity/wwwroot/static/index.css:152]: EOF
